auth/main_test.go

```diff
@@ -0,0 +1,103 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+
+	"time"
+
+	"github.com/gin-contrib/cors"
+	"github.com/gin-gonic/gin"
+	"github.com/stretchr/testify/assert"
+
+	"net/http/httptest"
+	"testing"
+)
+
+func setupRouter() *gin.Engine {
+	r := gin.Default()
+
+	r.Use(cors.New(cors.Config{
+		AllowOrigins:     []string{"http://localhost:5173"},
```

Comment:
Завязываться на одном порту не очень хорошо, лучше написать нечто подобное для локальной разработки
```suggestion
		AllowAllOrigins:     true,
```